server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
One more note - it is possible to set the write_concern in one place, so that we do not have to configure it on every single query. For example, we could do it on the MongoClient when we make the connection:

http://api.mongodb.org/python/current/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient

I believe you can also set the write_concern when you create the Collection instance.